### DISCUSSION TITLE
Why does addons look like they do?

### DISCUSSION THREAD

**Discussion by DROIDshield** [umbrellaplug/umbrellaplug.github.io#199] [Q&A]
Weird question i know, but rather than load any android box or pc with two addons to make the look of the addon doing the work look pleasing to the eye.

So im using AH2 for the look and navigation ease but running umbrella to do the work.

Isn't it easier to develop the addon already skinned to emulate the look of Netflix or some other. This then all works together under one install?

What's the reason no one developer seem to do this... ?im just puzzled.

**Comment by umbrellaplug** [comment]
Addons and Skins are different. Skins have access to things in Kodi that addons do not. Addons can do things and call functions skins cannot. They really are just different projects. One example would be "fonts", we haven't found any way to add a custom font to an addon, but a skin can add a custom font no problem. We can use a font from the active skin, but cannot bring in our own. That's just one example.

Skins are a lot of work. If you're not a "graphics and design" developer, it's not really a task you would want to take on, but you still might develop an addon.

Could more be done to change the designs around the addons where they aren't just lists? Sure... a lot of them do include different viewtypes for you so can get a little more of the look you want.

A lot of respect from me to the skin developers out there, especially Jurial and AH2. They are really pushing the border of what a skin should be doing and what the addon should be doing. You end up with some of the overlaps we have now where widgets are being made on seasons view that call episodes but are listed as tvshows. It's impressive, but it's a ton of work.

Consider it like this. A skin would be 1 project and an addon would be 1 project. For an addon developer to want to do skins, they would need to do both addons and skins. Example being AH2 with TMDBHelper. TMDBHelper is an addon. Why would he build an addon also instead of just building those things into ah2? The answer is because he had to in order to get the functionality he needed.

Now here is my personal thoughts on it..
I've tried to make Umbrella look better. I get it, it's not pretty. I've worked on using artwork in more places, adding options for different views, adding more options for colors, adding more options for scraping views. The most recent update added the option add icon packs from a zip file. Anyone could take the example template from here and make their own pack of icons, send it to me, and it be available in the plugin within hours for anyone using Umbrella. Since doing that, I have had **ZERO** requests. After a bit of time doing this I'm finding people want everything, but they don't want to put in any effort at all. I have a hard time getting people to send logs when they report an issue. Thinking anyone was going to put in the effort to create their own icon packs... a total waste. I'm not a designer, I'm a developer. I like to write code, I hate making graphics and suck at it. I've tried to get graphic help from the community and that was just another total waste of time.

**Reply by DROIDshield** [reply]
Well I have to say that was an overview I was not expecting, but has now given me a better understanding of how the two projects come together. 

Well I have used you new icon pack setting and have moved away from the default as soo as you had put it out there.

What I didn't know is a user has the chance now to make there own icon package.  I wouldn't have a clue how it's done like your coding when I see that I just think, what am I looking at here. It's  not an avenue I could go down although am so fascinated but this work and spend many hours tweeking my setup... either in the living room or in the bedroom as I have a shield in both. Infact some nights I don't even go to bed I just tweek until the morning like now at 2.40am still at it just done the latest AH2 update. 

I understand your frustration as to what users want and give nothing or very little back.

All I can say for myself is if I had the knowledge you have in coding or graphics... I'd spend my time trying to make the system better and better and better.

But for sure your work has gone far and wide and is respected the world over.... you must know this, but for sure if your not getting back from the community what you hoped.. be boyed up by the fact we do know you are putting in the hours for free and for all of our selfish use.

Again trust me when I say you are totally and fully respected. And I for one salute you sir.

Point me to the ware I can buy you a coffee because I am happy to donate. And here's your work in action many miles away in Sheffield UK.
![20230118_023820](https://user-images.githubusercontent.com/97023369/213068571-7b21ff82-51a2-4ffc-9094-06f5419f1cce.jpg)
